### PR TITLE
Port changes from basho/riak into node_package

### LINKED ISSUE #10: LOGNAME incorrectly used in startup scripts
Bring the following changes into node_package
- https://github.com/basho/riak/commit/176bb0f6a1fe2d65b20aea09842ebda6d51a030f
- https://github.com/basho/riak/commit/46636000a5263d6f51195afcf3574e8fbd934a50

### PR DESCRIPTION
**Pull request by jaredmorrow**
# Changes

Many bug fixes and useability changes have happened in Riak
that needed to be copied over to node_package. The major change
was to specify `/bin/sh` as the shell to enable node_package
use on systems which don't ship with `bash` namely Solaris.

In addition, node_package used LOGNAME in its scripts which
was found to be unreliable across platforms in Riak. Changes
were made to use whois instead to fix #10.
# Bugs / Issues Fixed

Fixes: #10 #12
# Testing

The package was tested by doing the following on an Ubuntu 12.04 clean install with Riak installed
- Change basho/stanchion `rebar.config` to point to branch `gh12-xplatform-support`
- `make package RELEASE=1`
- Make sure `riak` is installed
- `sudo dpkg -i <stanchion pkg>`
- Ensure the following work
  - `stanchion start`
  - `stanchion ping`
  - ensure `stanchion console` errors and tells you to `stanchion attach`
  - `stanchion attach`
  - `stanchion stop`
  - ensure `stanchion ping` fails
  - change the `/etc/stanchion/app.config` to have a syntax error (I added a "." randomly)
  - ensure `stanchion chkconfig` and `stanchion start` both error out
# Known Issues
- `appname version` does not work yet
- `appname reboot` does not work, see #16

### CODE REVIEW

File: priv/base/runner [hunk 1]
```diff
@@ -57,40 +83,64 @@ case "$1" in
...
+        ping_node
+        ES=$?
+        if [ "$ES" -ne 0 ]; then
+            exit $ES
+        fi
         ;;
 
     attach)
-        node_up_check
+        # Allow attaching to a node without pinging it
```
Comment (kellymclaughlin) [inline priv/base/runner:111]: This is terrific!

File: priv/base/env.sh [hunk 2]
```diff
@@ -52,36 +73,65 @@ ERTS_PATH=$RUNNER_BIN_DIR/erts-$ERTS_VSN/bin
...
-        else
-            echo "You must be $RUNNER_USER or root to invoke this script!"
+
+    if ([ "$RUNNER_USER" ] && [ "x$WHOAMI" != "x$RUNNER_USER" ]); then
+        type sudo > /dev/null 2>&1
+        if [ $? -ne 0 ]; then
+            echoerr "sudo doesn't appear to be installed and your EUID isn't $RUNNER_USER" 1>&2
             exit 1
         fi
+        echo "Attempting to restart script through sudo -H -u $RUNNER_USER"
```
Comment (kellymclaughlin) [inline priv/base/env.sh]: Is this valuable output? I can see how it would be useful if there are problems, but it's noisy for normal usage.

Comment (jaredmorrow) [inline priv/base/env.sh]: I was just following the Riak way, but yeah I'm not the biggest fan of it either.   I can remove it.

**Comment by kellymclaughlin** [comment]
The packaging errors on uninstall on Ubuntu seem to be resolved now.

**Comment by kellymclaughlin** [comment]
Did a test with `riak-cs` and `stanchion` and everything looks good. I'll leave it up to you if you want to remove the output from my previous comment. Otherwise, +1 to merge.

**Pull request merged**